app/controllers/middleware_server_controller.rb

```diff
@@ -28,9 +28,69 @@ def listicon_image(item, _view)
     "svg/#{icon}.svg"
   end
 
+  def button
+    super unless %w(middleware_server_reload middleware_server_stop).include?(params[:pressed])
+
+    operation = params[:pressed]
+    items = identify_selected_servers
+
+    return if items.nil? # TODO: re-route somewhere?
+
+    run_server_operation(operation, items)
+
+    render :update do |page|
+      page.replace("flash_msg_div", :partial => "layouts/flash_msg")
+    end
+  end
+
   private ############################
 
   def display_name
     _('Middleware Servers')
   end
+
+  def identify_selected_servers
```

Comment:
can you explain the logic in this method? or add a comment there?
the last line regarding identify_record taking an id from params passed to controller versus returning items from another parameter miq_grid_checks confuses me.